backend/benefit/applications/api/v1/serializers.py

```diff
@@ -67,21 +67,25 @@
 from users.utils import get_company_from_request, get_request_user_from_context
 
 
-class ApplicationBasisSerializer(serializers.ModelSerializer):
+class DynamicFieldsModelSerializer(serializers.ModelSerializer):
     """
-    Only the unique identifier is exposed through the API.
+    A ModelSerializer that takes an additional `fields` argument that
+    controls which fields should be displayed.
     """
 
-    class Meta:
-        model = ApplicationBasis
-        fields = [
-            "identifier",
-        ]
-        extra_kwargs = {
-            "identifier": {
-                "help_text": "Unique slug that identifies the basis",
-            }
-        }
+    def __init__(self, *args, **kwargs):
```

Comment:
A possible alternative:

```python
def __init__(self, *args, **kwargs):
	super().__init__(*args, **kwargs)

	fields = self.context.get("fields", [])
	exclude_fields = self.context.get("exclude_fields", [])

	existing = set(self.fields)
	allowed = set(fields) - set(exclude_fields)
	for field_name in existing - allowed:
		self.fields.pop(field_name)
```

Reply:
After some thinking, I realized that the handler and applicant serializers do offer a different set of fields, so based on the discussion, I'll go with your suggestion.

Reply:
I changed it so that an empty "fields" argument is treated as if no fields were specified